docs/dev_guide/design_doc.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: default
+title: Design Documents
+parent: Developer Guides
+permalink: /docs/dev_guide/design_doc
+---
+# Design Documents
+
+## Venice Improvement Proposal (VIP)
+
+Venice follow the Venice Improvement Proposal (VIP). VIP is the mechanism used to propose changes to the 
+Venice Codebase. 
+
+Not all the changes require a VIP treatment, Please use your commonsense to check if writing this VIP help yourself 
+and the reviewers time instead of just doing it in the pull request itself. Generally, this applies to
+* Large code refactors
+* New features
```

Comment:
Although it may be a bit of a tongue twister, I prefer that we use "functionality" rather than "feature", since the latter is ambiguous in the context of machine learning, which a lot of Venice use cases fall into.

Reply:
That is a good point. s/feature/functionality applied.